projects/components/src/table/table.component.ts

```diff
@@ -731,6 +734,7 @@ export class TableComponent
   private updateVisibleColumns(visibleColumnConfigs: TableColumnConfigExtended[]): void {
     this.visibleColumnConfigs = visibleColumnConfigs;
     this.visibleColumnIds = this.visibleColumnConfigs.map(column => column.id);
+    this.visibleColumnsChange.next(this.visibleColumnConfigs.filter(column => column.id !== '$$expanded'));
```

Comment:
why we're excluding only the expanded column? We also have another state column, we can use `isStateColumn` function to check

Reply:
thanks, good catch